src/apps/replication/meta_server/meta_service.cpp

```diff
@@ -222,7 +222,7 @@ void meta_service::on_query_configuration_by_node(dsn_message_t msg)
 
 void meta_service::on_query_configuration_by_index(dsn_message_t msg)
 {
-    if (!check_primary(msg))
+    if (_failure_detector == nullptr || !check_primary(msg))
```

Comment:
instead of fixing this way, let's move "_failure_detector = new meta_server_failure_detector(_state, this);" before rpc_regiser in meta_service.